### PR TITLE
Java: Add the `XGROUP CREATE` and `XGROUP DESTROY` commands

### PR DESCRIPTION
**Pull request by acarbonetto**
https://valkey.io/commands/xgroup-create/
https://valkey.io/commands/xgroup-destroy/

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 1]
```diff
@@ -312,4 +317,59 @@ CompletableFuture<Map<String, String[][]>> xrevrange(
...
+     * @param key The key of the stream.
+     * @param groupname The newly created consumer group name.
+     * @param id Stream ID that specifies the last delivered entry in the stream from the new group’s
+     *     perspective. The special ID <code>"$"</code> can be used to specify the last entry in the
+     *     stream.
+     * @return <code>OK</code>.
+     * @example
+     *     <pre>{@code
+     * // Create the consumer group "mygroup", using zero as the starting ID:
+     * assert client.xgroupCreate("mystream", "mygroup", "0-0").get() == "OK";
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:335]: Replace `==` with `equals`

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 2]
```diff
@@ -312,4 +317,59 @@ CompletableFuture<Map<String, String[][]>> xrevrange(
...
+     * @param groupname The newly created consumer group name.
+     * @param id Stream ID that specifies the last delivered entry in the stream from the new group’s
+     *     perspective. The special ID <code>"$"</code> can be used to specify the last entry in the
+     *     stream.
+     * @param options The group options {@link StreamGroupOptions}.
+     * @return <code>OK</code>.
+     * @example
+     *     <pre>{@code
+     * // Create the consumer group "mygroup", and the stream if it does not exist, after the last ID
+     * assert client.xgroupCreate("mystream", "mygroup", "$", StreamGroupOptions.builder().makeStream(true).build()).get() == "OK";
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:355]: same

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 3]
```diff
@@ -312,4 +317,59 @@ CompletableFuture<Map<String, String[][]>> xrevrange(
...
+     * Destroys the consumer group <code>groupname</code> for the stream stored at <code>key</code>.
+     *
+     * @see <a href="https://valkey.io/commands/xgroup-destroy/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @param groupname The newly created consumer group name.
+     * @return <code>true</code> if the consumer group is destroyed. Otherwise, <code>false</code>.
+     * @example
+     *     <pre>{@code
+     * // Destroys the consumer group "mygroup"
+     * assert client.xgroupDestroy("mystream", "mygroup").get() == "OK";
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:371]: same

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 4]
```diff
@@ -312,4 +317,59 @@ CompletableFuture<Map<String, String[][]>> xrevrange(
      */
     CompletableFuture<Map<String, String[][]>> xrevrange(
             String key, StreamRange end, StreamRange start, long count);
+
+    /**
+     * Create a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:322]:
```suggestion
     * Creates a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 5]
```diff
@@ -312,4 +317,59 @@ CompletableFuture<Map<String, String[][]>> xrevrange(
...
+     * @example
+     *     <pre>{@code
+     * // Create the consumer group "mygroup", using zero as the starting ID:
+     * assert client.xgroupCreate("mystream", "mygroup", "0-0").get() == "OK";
+     * }</pre>
+     */
+    CompletableFuture<String> xgroupCreate(String key, String groupname, String id);
+
+    /**
+     * Create a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:341]:
```suggestion
     * Creates a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 6]
```diff
@@ -2845,6 +2852,61 @@ public T xrevrange(
         return getThis();
     }
 
+    /**
+     * Create a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2856]:
```suggestion
     * Creates a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 7]
```diff
@@ -2845,6 +2852,61 @@ public T xrevrange(
...
+     *     stream.
+     * @return Command Response - <code>OK</code>.
+     */
+    public T xgroupCreate(@NonNull String key, @NonNull String groupname, @NonNull String id) {
+        protobufTransaction.addCommands(buildCommand(XGroupCreate, buildArgs(key, groupname, id)));
+        return getThis();
+    }
+
+    /**
+     * Create a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2873]:
```suggestion
     * Creates a new consumer group uniquely identified by <code>groupname</code> for the stream stored
```

File: java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+import lombok.Builder;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xgroupCreate(String, String, String,
+ * StreamGroupOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/xgroup-create/">valkey.io</a>
+ */
+@Builder
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:15]: I think user API with 3 constructors or factory methods will be easier

File: java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:1]: Please confirm that this class is exposed to the user

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:1]: Yes.  stream options have all been added to the JAR module.

File: java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java (hunk 8 above)
Comment (acarbonetto) [reply java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:15]: Are you suggesting: 
```
xgroupCreate(String key, String groupname, String id)
xgroupCreate(String key, String groupname, String id, boolean makeStream)
xgroupCreate(String key, String groupname, String id, String entriesRead)
xgroupCreate(String key, String groupname, String id, boolean makeStream, String entriesRead)
```

Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:15]:
```java
new StreamGroupOptions(bool)
new StreamGroupOptions(bool, String)
new StreamGroupOptions(String)
```

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:15]: ok...

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java (hunk 1 above)
Comment (acarbonetto) [reply java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:335]: right

File: java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+    public StreamGroupOptions(String entriesRead) {
+        this.makeStream = false;
+        this.entriesRead = entriesRead;
+    }
+
+    public StreamGroupOptions(Boolean makeStream, String entriesRead) {
+        this.makeStream = makeStream;
+        this.entriesRead = entriesRead;
+    }
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java:43]: Please add descriptions to the constructors. These do not require complete docs.
You can move/copy the from the member declarations.

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**